lambeq/rewrite/base.py

```diff
@@ -393,3 +399,16 @@ def _ar(self, box: Box) -> Diagram:
 
     def _ob(self, ob: Ty) -> Ty:
         return ob
+
+
+class ImperativeRewriteRule():
+    """A rewrite rule for imperative sentence diagrams.
+    """
+    def matches(self, diagram: Diagram) -> bool:
+        return diagram.cod == N.r @ S
+
+    def rewrite(self, diagram: Diagram) -> Diagram:
+        if self.matches(diagram):
+            return (Box('IMP', Ty(), N) @ diagram >> Cup(N, N.r) @ Id(S))
```

Comment:
Another way of doing it is to create a Box that has the `diagram.cod` as `dom` and `S` as `cod` and then return `diagram >> box`